src/engine/pieces/piece.ts

```diff
@@ -13,8 +17,35 @@ export default class Piece {
         throw new Error('This method must be implemented, and return a list of available moves');
     }
 
+    public addAvailableMoveAndContinue(availableMoves: Array<Square>, newRow : number, newCol : number, currentPosition : Square, board : Board){
+        let newPos = new Square(newRow, newCol);
+        let newPosInBound = newPos.inBoundsCheck();
+        if(!currentPosition.equals(newPos) && newPosInBound){
+            let pieceAtNewPos = board.getPiece(newPos);
+            if(pieceAtNewPos == undefined || (pieceAtNewPos.player != this.player && !pieceAtNewPos.isKing)){
+                availableMoves.push(newPos);
+            }
+            return pieceAtNewPos == undefined;
+        }
+        return newPosInBound;
```

Comment:
These return values feel a bit unexpected from the function name

Also we want to avoid exposing unecessary variables to functions where possible. I think here you could refactor slightly to not pass in availableMoves, as this function should not need to know the existing available moves.

You could then have it return the new move instead of 
`availableMoves.push(newPos);`

and externally add this to `availableMoves`

Reply:
You then shouldn't need to return anything else

Reply:
The return value is to see whether to break our path check e.g. a piece is in the way. We changed the function to be called addAvailableMoveAndCheckContinue to make it clearer that we are checking whether to continue or not

Reply:
I was thinking if you returned the `newPos `if there was one that would solve the same issue, can chat in person